ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -138,144 +153,241 @@ public:
     if (numberOfEASParameters != 0)
       easApplicabilityCheck();
     easVariant_.setEASType(numberOfEASParameters);
+    initializeState();
   }
 
+  /**
+   * \brief Gets the internal state variable alpha for the EAS element.
+   *
+   * \return Internal state variable (alpha).
+   */
+  const auto& alpha() const { return alpha_; }
+
 protected:
   void bindImpl() {
     assert(underlying().localView().bound());
     easVariant_.bind(underlying().localView().element().geometry());
+    initializeState();
+  }
+
+  /**
+   * \brief Updates the internal state variable alpha_ at the end of an iteration before the update of the displacements
+   * done by the non-linear solver. See \cite bieberLockingHourglassingNonlinear2024 for implementation details and
+   * further references.
+   *
+   * \param par The Requirement object.
+   * \param correction The correction in displacement (DeltaD) vector passed based on which the internal state variable
+   * alpha is to be updated.
+   */
+  void updateStateImpl(const Requirement& par,
+                       const std::remove_reference_t<typename Traits::template VectorType<>>& correction) const {
```

Comment:
Is this a problem to have this method non const? It is still strange to have a method updateStateImpl being const... But if it is too difficult we can keep it.